mungegithub/mungers/dibs-nack-handler.go

```diff
@@ -0,0 +1,195 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"strings"
+
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/contrib/mungegithub/github"
+	e "k8s.io/contrib/mungegithub/mungers/matchers/event"
+	"k8s.io/contrib/mungegithub/mungers/mungerutil"
+
+	"github.com/golang/glog"
+	goGithub "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+const OWNER = "kubernetes"
+
+// DibsNackHandler will
+// - apply LGTM label if reviewer has commented "/lgtm", or
+// - remove LGTM label if reviewer has commented "/lgtm cancel"
+type DibsNackHandler struct {
+	features *features.Features
+}
+func init() {
+	l := DibsNackHandler{}
+	RegisterMungerOrDie(l)
+}
+
+// Name is the name usable in --pr-mungers
+func (DibsNackHandler) Name() string { return "dibs-nack-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (DibsNackHandler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (DibsNackHandler) Initialize(config *github.Config, features *features.Features) error { return nil }
+
+// EachLoop is called at the start of every munge loop
+func (DibsNackHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (DibsNackHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (h DibsNackHandler) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+	reviewers := getReviewers(obj)
+	if len(reviewers) == 0 {
+		return
+	}
+
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Errorf("unexpected error getting comments: %v", err)
+		return
+	}
+
+	events, err := obj.GetEvents()
+	if err != nil {
+		glog.Errorf("unexpected error getting events: %v", err)
+		return
+	}
+
+	h.assignIfDibs(obj, comments, events, reviewers)
+	h.removeIfNack(obj, comments, events, reviewers)
+}
+
+func (h *DibsNackHandler) assignIfDibs(obj *github.MungeObject, comments []*goGithub.IssueComment, events []*goGithub.IssueEvent, reviewers mungerutil.UserSet) {
+	// Get the last time when the someone applied lgtm manually.
+	removeLGTMTime := e.LastEvent(events, e.And{e.RemoveLabel{}, e.LabelName(lgtmLabel), e.HumanActor()}, nil)
+
+	// Assumption: The comments should be sorted (by default from github api) from oldest to latest
+	// If they, dibs then nack. Remove them.
+
+	for i := len(comments) - 1; i >= 0; i-- {
```

Comment:
I find it hard to read and/or understand what it does. Let's see how the discussion about `/reassign` changes that before going into further details